app/upgrades.go

```diff
@@ -1,3 +1,188 @@
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
+	}
+}
+
+func UpdateCosmosMintInflation(ctx sdk.Context, mintKeeper mintkeeper.Keeper) {
+	mintParams := mintKeeper.GetParams(ctx)
+	// 0.75
+	mintParams.InflationMin = sdk.NewDecWithPrec(75, 2)
+	mintParams.InflationMax = sdk.NewDecWithPrec(75, 2)
+
+	mintKeeper.SetParams(ctx, mintParams)
+}
+
+func UpdateSavingsParams(ctx sdk.Context, savingsKeeper savingskeeper.Keeper) {
+	savingsParams := savingstypes.NewParams([]string{
+		"ukava",
+		"bkava",
+		"erc20/multichain/usdc",
+	})
+
+	savingsKeeper.SetParams(ctx, savingsParams)
+}
+
+func UpdateEvmutilPermissions(ctx sdk.Context, accountKeeper authkeeper.AccountKeeper) {
+	// add minter and burner permissions to evmutil
+	evmutilAcc, ok := accountKeeper.GetModuleAccount(ctx, evmutiltypes.ModuleName).(*authtypes.ModuleAccount)
+	if !ok {
+		panic("unable to fetch evmutil module account")
+	}
+	evmutilAcc.Permissions = []string{
+		authtypes.Minter, authtypes.Burner,
+	}
+	accountKeeper.SetModuleAccount(ctx, evmutilAcc)
+}
+
+//go:embed eth_eoa_addresses.json
+var ethEOAAddresses []byte
+
+func IterateEOAAddresses(f func(addr string)) {
+	var addresses []string
+
+	if err := json.Unmarshal(ethEOAAddresses, &addresses); err != nil {
+		panic("failed to unmarshal embedded eth_eoa_addresses.json")
+	}
+
+	for _, addr := range addresses {
+		f(addr)
+	}
+}
+
+// ConvertEOAsToBaseAccount converts all non-contract EthAccounts to BaseAccounts
+func ConvertEOAsToBaseAccount(ctx sdk.Context, accountKeeper authkeeper.AccountKeeper) {
+	IterateEOAAddresses(func(addrStr string) {
+		addr, err := sdk.AccAddressFromBech32(addrStr)
+		if err != nil {
+			panic("failed to parse address")
+		}
+
+		// Skip non-EthAccounts
+		acc := accountKeeper.GetAccount(ctx, addr)
+		ethAcc, isEthAcc := acc.(*etherminttypes.EthAccount)
+		if !isEthAcc {
+			return
+		}
+
+		// Skip contract accounts
+		if ethAcc.Type() != etherminttypes.AccountTypeEOA {
+			return
+		}
+
+		// Change to BaseAccount in store
+		accountKeeper.SetAccount(ctx, ethAcc.BaseAccount)
+	})
+}
+
+func UpdatePricefeedParams(ctx sdk.Context, pricefeedKeeper pricefeedkeeper.Keeper) {
+	params := pricefeedKeeper.GetParams(ctx)
+	oracles := params.Markets[0].Oracles
+	newMarkets := pricefeedtypes.Markets{
+		{
+			MarketID:   "usdc:usd",
+			BaseAsset:  "usdc",
+			QuoteAsset: "usd",
+			Oracles:    oracles,
+			Active:     true,
+		},
+		{
+			MarketID:   "usdc:usd:30",
+			BaseAsset:  "usdc",
+			QuoteAsset: "usd",
+			Oracles:    oracles,
+			Active:     true,
+		},
+		{
+			MarketID:   "usdt:usd:30",
+			BaseAsset:  "usdt",
+			QuoteAsset: "usd",
+			Oracles:    oracles,
+			Active:     true,
+		},
+		{
+			MarketID:   "usdt:usd:30",
+			BaseAsset:  "usdt",
+			QuoteAsset: "usd",
+			Oracles:    oracles,
+			Active:     true,
+		},
+		{
+			MarketID:   "dai:usd:30",
+			BaseAsset:  "dai",
+			QuoteAsset: "usd",
+			Oracles:    oracles,
+			Active:     true,
+		},
```

Comment:
Duplicate market IDs with the TWAP :30 ones, fails when setting params.

`panic: value from ParamSetPair is invalid: duplicated market usdt:usd:30`

Reply:
:+1:, fixed the bad markets